src/utils/MainApi.js

```diff
@@ -0,0 +1,82 @@
+import { URL_MAIN } from './constants'
+
+class MainApi {
+  constructor(options) {
+    this._baseUrl = options.baseUrl
+    this._headers = options.headers
+  }
+
+  _checkResponse(res) {
+    if (res.ok) {
+      return res.json()
+    }
+    return Promise.reject(`Error ${res.status}`)
+  }
+
+  validateToken(token) {
+    return fetch(this._baseUrl + '/users/me', {
+      headers: { ...this._headers, authorization: 'Bearer ' + token },
+    }).then(this._checkResponse)
```

Comment:
**Might be better**

You can make a special method for fetching and checking responses **not to duplicate** it in every request:

```js

  _request(url, options) {
    return fetch(url, options).then(this._checkResponse)
  }

```

And now you can safely replace `fetch` with `this._request`. And you don’t need to change anything else.